Panic on decode squashed in squashed struct

Run of such test panics.
```go
func TestNextSquashMapstructure(t *testing.T) {
	data := &struct {
		Level1 struct {
			Level2 struct {
				Foo string
			} `mapstructure:",squash"`
		} `mapstructure:",squash"`
	}{}
	err := mapstructure.Decode(map[interface{}]interface{}{"foo": "baz"}, &data)
	require.NoError(t, err)
	assert.Equal(t, "baz", data.Level1.Level2.Foo)
}
```
```
panic: reflect: call of reflect.Value.Type on zero Value

panic(0x2f5d80, 0xc4202d4ae0)
	1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc4200a0240)
	1.7.3/libexec/src/testing/testing.go:579 +0x25d
panic(0x2f5d80, 0xc4202d4ae0)
	1.7.3/libexec/src/runtime/panic.go:458 +0x243
reflect.Value.Type(0x0, 0x0, 0x0, 0x2c53b1, 0x6)
	1.7.3/libexec/src/reflect/value.go:1670 +0x224
mitchellh/mapstructure.(*Decoder).decodeStruct(0xc420084268, 0x0, 0x0, 0x2f9080, 0xc420247d10, 0x300d00, 0xc4202ad110, 0x199, 0x0, 0x0)
	mitchellh/mapstructure/mapstructure.go:672 +0x625
mitchellh/mapstructure.(*Decoder).decode(0xc420084268, 0x0, 0x0, 0x2f9080, 0xc420247d10, 0x300d00, 0xc4202ad110, 0x199, 0xc41fffaa0c, 0x0)
	mitchellh/mapstructure/mapstructure.go:225 +0x6e8
mitchellh/mapstructure.(*Decoder).decodePtr(0xc420084268, 0x0, 0x0, 0x2f9080, 0xc420247d10, 0x2d67c0, 0xc420084260, 0x196, 0xc42008cde0, 0x1122e)
	mitchellh/mapstructure/mapstructure.go:557 +0x15b
mitchellh/mapstructure.(*Decoder).decode(0xc420084268, 0x0, 0x0, 0x2f9080, 0xc420247d10, 0x2d67c0, 0xc420084260, 0x196, 0xc4202ad101, 0xc420284fc0)
	mitchellh/mapstructure/mapstructure.go:229 +0x617
mitchellh/mapstructure.(*Decoder).Decode(0xc420084268, 0x2f9080, 0xc420247d10, 0x0, 0xc4202ad130)
	mitchellh/mapstructure/mapstructure.go:180 +0xb3
mitchellh/mapstructure.Decode(0x2f9080, 0xc420247d10, 0x2cb000, 0xc420084260, 0xc4202ad130, 0x0)
	mitchellh/mapstructure/mapstructure.go:122 +0xb6
TestNextSquashMapstructure(0xc4200a0240)
```

--- a/mapstructure.go
+++ b/mapstructure.go
@@ -712,7 +712,7 @@ func (d *Decoder) decodeStruct(name string, data interface{}, val reflect.Value)
 					errors = appendErrors(errors,
 						fmt.Errorf("%s: unsupported type for squash: %s", fieldType.Name, fieldKind))
 				} else {
-					structs = append(structs, val.FieldByName(fieldType.Name))
+					structs = append(structs, structVal.FieldByName(fieldType.Name))
 				}
 				continue
 			}

--- a/mapstructure_bugs_test.go
+++ b/mapstructure_bugs_test.go
@@ -258,3 +258,21 @@ func TestDecodeSliceToEmptySliceWOZeroing(t *testing.T) {
 		}
 	}
 }
+
+// #70
+func TestNextSquashMapstructure(t *testing.T) {
+	data := &struct {
+		Level1 struct {
+			Level2 struct {
+				Foo string
+			} `mapstructure:",squash"`
+		} `mapstructure:",squash"`
+	}{}
+	err := Decode(map[interface{}]interface{}{"foo": "baz"}, &data)
+	if err != nil {
+		t.Fatalf("should not error: %s", err)
+	}
+	if data.Level1.Level2.Foo != "baz" {
+		t.Fatal("value should be baz")
+	}
+}

